test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10589,6 +10589,535 @@ BOOST_AUTO_TEST_CASE(snark)
+			return a;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(fixed128x19)", [](u256 const &a) -> u256 { return a; }, -50, -1);
+	testContractAgainstCppOnRange("f(fixed128x19)", [](u256 const &a) -> u256 { return a; }, 0, 50);
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_addition)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (ufixed) {
+			return 1.25 + 2.5;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(375, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_constant_addition)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (fixed) {
+			return -1.25 + -2.5;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-375, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_addition)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 1.0;
+		ufixed b = 2.5;
+		ufixed c = 1.25;
+
+		function f() public returns (ufixed) {
+			return a + b + c;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(475, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_addition)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = -1.0;
+		fixed b = -2.5;
+		fixed c = -1.25;
+
+		function f() public returns (fixed) {
+			return a + b + c;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-475, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_subtraction)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (ufixed) {
+			return 3.5 - 1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_constant_subtraction)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f() public returns (fixed) {
+			return 1.25 - 3.5;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_subtraction)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (ufixed) {
+			return a - b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_subtraction)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = 3.5;
+		fixed b = 1.25;
+
+		function f() public returns (fixed) {
+			return b - a;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-225, 100, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_multiplication)
```

Comment:
This test actually does have no relation to fixed points, it is about rationals.

Reply:
Yeah, I realized that. The same goes for all of the other constant arithmetic tests.